### ISSUE TITLE
mill.scalalib.GenIdea/idea fails with netty-transport-native-epoll as dependency

### ISSUE THREAD

**Issue by kxa14** [com-lihaoyi/mill#1188]
`mill.scalalib.GenIdea/idea` fails with the following error when executing on a project that has netty-transport-native-epoll as a dependency. 

```
1 targets failed
mill.scalalib.GenIdea.idea java.nio.file.NoSuchFileException: ~/.cache/coursier/v1/https/repo1.maven.org/maven2/io/netty/netty-transport-native-epoll/4.1.46.Final/netty-transport-native-epoll-4.1.46.Final-linux-x86_64.pom
```

It's complaining about a missing pom file which has a `linux-x86_64` suffix. Looking into the `~/.cache/coursier.../4.1.46.Final` directory, we can see the said pom file does not exists, whereas a pom file without the suffix exists. 

```
 ~/coursier/v1/https/repo1.maven.org/maven2/io/netty/netty-transport-native-epoll/4.1.46.Final:
   netty-transport-native-epoll-4.1.46.Final-linux-x86_64.jar
   netty-transport-native-epoll-4.1.46.Final-sources.jar
   netty-transport-native-epoll-4.1.46.Final.pom 
```

The issue seems to come from the logic in [GenIdeaImpl](https://github.com/com-lihaoyi/mill/blob/6cf96e4b01acd78ff31eed800b4e224d7a425998/scalalib/src/GenIdeaImpl.scala#L333) looks for a POM extension with the same filename as the jar which does not cater for classifiers. 

We think one solution to resolve this issue is to change the code to look for any file with a POM extension in the same directory as the jar.

More detailed information by @zainab-ali can be found here https://github.com/lsug/lsug-website/issues/90

Would you like us to PR this change? Also, thank you for working on and maintaining this build tool. Our project has benefited a lot from this project. :slightly_smiling_face:

**Comment by lefou** [comment]
Can you show how you defined the dependency? Did you used the classifier? In case you did not used a proper classifier, you should first try if using it fixes the issue. Something like this: 

```scala
ivy"io.netty:netty-transport-native-epoll:4.1.46.Final;classifier=linux-x86_64"
```

If that doesn't help, yeah, a PR would be great. But the proper name of the pom file should be calculated from the maven/coursier coordinates. Picking random pom files will open more issues eventually instead of fixing them.

**Comment by lefou** [comment]
@kxa14 Could you try this branch? https://github.com/lefou/mill/tree/idea-sources 

The issue is, that we, in the place where the idea generator fails, only see jars but don't have their ivy dependencies at hand, so we must play a guess game to find the source jars. In this branch I tried to guess a little bit better.

**Comment by lefou** [comment]
I just merged PR #1195.

@kxa14 Your feedback would be much appreciated.

**Comment by kxa14** [comment]
@lefou Sorry for the late reply. Yes, it worked when I ran `mill.scalalib.GenIdea/idea` with the fix. I can see on the log file that it's parsing the right pom file 

```
Parsing POM /home/kaixin/.cache/coursier/v1/https/repo1.maven.org/maven2/io/netty/netty-transport-native-epoll/4.1.46.Final/netty-transport-native-epoll-4.1.46.Final.pom ...
```
Thank you for looking into this.

**Comment by lefou** [comment]
Fixed by #1195.
